controllers/upgrade/premisesChecker_test.go

```diff
@@ -0,0 +1,218 @@
+		fromDoguResource := readTestDataRedmineCr(t)
+		fromDogu := readTestDataDogu(t, redmineBytes)
+		toDogu := readTestDataDogu(t, redmineBytes)
+
+		mockedChecker := new(premiseMock)
+		mockedChecker.On("CheckWithResource", fromDoguResource).Return(errors.New("CheckWithResource"))
+
+		sut := NewPremisesChecker(mockedChecker, mockedChecker, mockedChecker)
+
+		// when
+		err := sut.Check(ctx, fromDoguResource, fromDogu, toDogu)
+
+		// then
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "CheckWithResource")
+		assert.IsType(t, &requeueablePremisesError{}, err)
+		// prove that the above negative type assertion works by positive type assertion
+		_, ok := err.(*requeueablePremisesError)
+		assert.True(t, ok)
+		mockedChecker.AssertExpectations(t)
+	})
+	t.Run("should fail when dogu health check fails", func(t *testing.T) {
+		fromDoguResource := readTestDataRedmineCr(t)
+		fromDogu := readTestDataDogu(t, redmineBytes)
+		toDogu := readTestDataDogu(t, redmineBytes)
+
+		mockedChecker := new(premiseMock)
+		mockedChecker.On("CheckWithResource", fromDoguResource).Return(nil)
+		mockedChecker.On("ValidateDependencies", fromDogu).Return(errors.New("ValidateDependencies"))
+
+		sut := NewPremisesChecker(mockedChecker, mockedChecker, mockedChecker)
+
+		// when
+		err := sut.Check(ctx, fromDoguResource, fromDogu, toDogu)
+
+		// then
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "ValidateDependencies")
+		assert.IsType(t, &requeueablePremisesError{}, err)
+		mockedChecker.AssertExpectations(t)
+	})
+	t.Run("should fail when dogu dependency health check fails", func(t *testing.T) {
+		fromDoguResource := readTestDataRedmineCr(t)
+		fromDogu := readTestDataDogu(t, redmineBytes)
+		toDogu := readTestDataDogu(t, redmineBytes)
+
+		mockedChecker := new(premiseMock)
+		mockedChecker.On("CheckWithResource", fromDoguResource).Return(nil)
+		mockedChecker.On("ValidateDependencies", fromDogu).Return(nil)
+		mockedChecker.On("CheckDependenciesRecursive", fromDogu, fromDoguResource.Namespace).Return(errors.New("CheckDependenciesRecursive"))
+
+		sut := NewPremisesChecker(mockedChecker, mockedChecker, mockedChecker)
+
+		// when
+		err := sut.Check(ctx, fromDoguResource, fromDogu, toDogu)
+
+		// then
+		require.Error(t, err)
+		assert.Contains(t, err.Error(), "CheckDependenciesRecursive")
+		assert.IsType(t, &requeueablePremisesError{}, err)
+		mockedChecker.AssertExpectations(t)
+	})
+}
+
+type premiseMock struct {
+	mock.Mock
+}
+
+func (pm *premiseMock) ValidateDependencies(_ context.Context, dogu *core.Dogu) error {
+	args := pm.Called(dogu)
+	return args.Error(0)
+}
+
+func (pm *premiseMock) CheckDependenciesRecursive(_ context.Context, fromDogu *core.Dogu, currentK8sNamespace string) error {
+	args := pm.Called(fromDogu, currentK8sNamespace)
+	return args.Error(0)
+}
+
+func (pm *premiseMock) CheckWithResource(_ context.Context, doguResource *k8sv1.Dogu) error {
+	args := pm.Called(doguResource)
+	return args.Error(0)
+}
```

Comment:
Maybe replace with a mock generated by `mockery` which is by far the more common in the operator.